### DISCUSSION TITLE
[Feature Request] HALF Z-HOP

### DISCUSSION THREAD

**Discussion by rhlacerda** [OrcaSlicer/OrcaSlicer#3439] [Ideas]
Hi! I have a K1, and I am printing some small models. I need to use Z-hop because my models have some small details, and when the hotend travels... sometimes hit and break the model.

I tried all types of hops and it leaves strings or some blobs on it. Lost a lot of time trying to configure retraction, temp and other stuffs to solve it... but when I turn off z-hop... I got a perfect and clean print... but I can't use it off, because I want to print 30 models at the same gcode, if the nozzle hit a model... GG gonna be a mess.

So I got one idea... What if we have a half z-hop process...
Will work this way:

1- the nozzle leave the model like z-hop is disabled;
2- moves up when the nozzle isn't over the print;

something like this:
___/

a mix of off z-hop + slope z-hop

I am using wipe with z-hop on and z-hop models. Tried without it too... but the best print has wipe without Z-hop.

Thanks! Hope it is possible and can help every maker =)

**Comment by rhlacerda** [comment]
Hi! Any one know if, is this possible?

**Comment by gsgsagaz** [comment]
It is a very good idea, it is the best zhop i ever see in scene.

**Comment by TonOnerd** [comment]
I believe that in this way, it can reduce the excess amount when the nozzle lifts, it would work as a cleaning before the retraction when leaving the part and when raising the Z avoids collision with the next point.

**Comment by MARCELOGSI** [comment]
Seria muito importante essa atualização, para peças menores e redução de strings.

**Comment by Con-Rado** [comment]
It would be a big improvement on Z-Hop Feature

**Comment by Rickthebig** [comment]
I am painfully missing this feature.
Improving the original idea: Option to choose whether the actual lift should occur before reaching the next outer wall or in mid air (but in any case before reaching the outer wall of the target zone).

Any updates here?

**Comment by Rickthebig** [comment]
To whom it may concern: There is a similar, although suboptimal solution here
https://teachingtechyt.github.io/diagonalZhop.html
Orca has a variety of different z hops, but all somehow miscarried. The method proposed here would be perfect, as simple as it is.

**Reply by rhlacerda** [reply]
I tested all types of z-hop available, and they all leave strings, especially on small details. But when I do the same print without using z-hop, there's no sign of stringing, but bullying occurs. My tests were on miniatures with very small details. Therefore, I think a good solution for z-hop would be...

Print, clean, leave like there is no z-hop, go up, go down, print, Repeat.

![Z-hop](https://github.com/user-attachments/assets/6da12f78-af09-4131-873a-5091d87bec81)
